Added text to be displayed when invalid data is entered into search input field and  no search results are found

san-coding opened the pull request:
Added text to be displayed when invalid data is entered into the search input field and no search results found
 in `WooCommerce-> Marketplace` , fixes issue #30588



### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Closes # .

### How to test the changes in this Pull Request:

1. Create any test site using JN site.
2. Install and activate all the required plugins.
3. Install and activate WooCommerce 5.7.0 Beta-1.
4. Complete the setup wizard.
5. Go to WooCommerce->Marketplace.
6. Search invalid data (non-existing) in search field.
7. Screen displays `Sorry, could not find anything. Try searching again using a different term.`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Tweak - Show a search again message when marketplace results are empty.

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

san-coding committed:
Added text to be displayed when no search results 

Added text to be displayed when no search results found , fixes issue #30588

san-coding commented:
Hey @vedanshujain @andfinally @kloon  , I have added this PR #30641 , please review it , thank you

san-coding commented:
Ok , now I see that the changes I have made are wrong, but how do I know that no search results are being returned

san-coding commented:
> now I see that the changes I have made are wrong, but how do I know that no search results are being returned , is any variable storing the number of results returned, please help me out @vedanshujain @andfinally @kloon , thanks

vedanshujain commented:
@san-coding you should be able to use `$addon` to see if there are search results. Can you check if the presence of both `$search` and more than one in `$addon` gives you expected behavior?

san-coding commented:
> @san-coding you should be able to use `$addon` to see if there are search results. Can you check if the presence of both `$search` and more than one in `$addon` gives you expected behavior?

Ok thanks , I will check and make changes

san-coding committed:
Merge branch 'woocommerce:trunk' into trunk

san-coding committed:
Displaying message if sizeof($addons)==0

made changes requested in PR #30641

san-coding committed:
Checking If sizeof($addons>0) 

Checking  sizeof($addons>0) along with !empty($search) to display search results if addons found

san-coding commented:
Hey @vedanshujain , could you please review this

san-coding committed:
Changed sizeof to count

san-coding commented:
Have added required space and replaced `sizeof()` with `count()`

san-coding commented:
@vedanshujain , looks like all checks have passed, you may review and merge

san-coding commented:
@vedanshujain @rodelgc @andfinally someone please review this

kloon commented on includes/admin/views/html-admin-page-addons.php:
```diff
@@ -62,7 +62,13 @@ class="<?php echo $current_section === $section->slug ? 'current' : ''; ?>"
-			<?php if ( ! empty( $search ) ) : ?>
+			<?php if ( count( $addons ) == 0 ) : ?>
+				<h1 class="search-form-title">
+					<?php // translators: search keyword. ?>
```
You can remove this since there is no placeholder in the string below.

san-coding replied:
should I make the below change ?
```suggestion
					<?php // translators: search keyword. ?>
```

san-coding replied:
What exactly should I remove

eopws replied:
He is talking about this `<?php // translators: search keyword. ?>`

san-coding replied:
have made the changes

san-coding committed:
made changes requested in PR

san-coding commented:
Thank you , I have made the changes, please review this PR

san-coding commented:
can you please add a label `hacktoberfest-accepted` to this PR, thank you

kloon reviewed (approved):
Changes look good to me now, thanks!

peterfabian commented:
Requesting a review from the core team before merging just to double check, should be a quick one.

vedanshujain reviewed (approved):
LGTM!

san-coding commented:
Thanks you 👍🏻
